docs/modules/ROOT/pages/explanations/decisions/postgres-monitoring.adoc

```diff
@@ -0,0 +1,65 @@
+= PostgreSQL by VSHN Availability check
+
+== Problem
+
+We need to monitor availability of PostgreSQL by VSHN to be able to measure "Up" https://products.docs.vshn.ch/products/appcat/postgresql.html#_service_level_indicator_sli[SLI]
+
+== Goals
+
+* measure Up SLI
+* idea how to integrate SLI https://products.docs.vshn.ch/products/service_levels.html#_exceptions_to_availability_guarantee[Exceptions]
+
+== Non Goals
+
+1. Collect any futher metrics
+2. SLO Definition
+3. Detailed concept of the SLI exceptions
+
+== Solutions
+
+=== Postgresql_exporter by Stackgres
+    
+    Stackgres provides us built-in postgresql_exporter running as a side car in database pod, which contains many useful metrics, but one especially usefull from availability monitoring perspective - pg_up. Pg_up is a metric created when all preconfigured checks SQL queries are properly executed and no one returned an error. Such metric could generate false positives in situation when one of the queries will be dropped by PostgreSQL. This exporter also connects locally, which makes such solution unreliable for us, because our customers connects in different way, connection has to pass through pg bouncer, it has to reuse user credentials and shouldn't generate false positives if one of two pods crash. It doesn't check for WRITE operations, only read operations.
+
+=== Postgresql_exporter by VSHN
+
+    We can also deploy our own instance of postgresql_exporter to database namespace. We'll receive the same amount of metrics, we'll also be able to measure uptime using pg_up metric, but in this solution we can configure exporter to use the same route, user, password and database as our customers use. It doesn't check for WRITE operations, only read operations.
+
+===  Customly written exporter
+
+    We can write our own exporter which we could run in database namespace. In such scenarion we can write minimalistic exporter that checks connectivity reusing customer connection string, can execute WRITE queries and it'll return for example pg_up metric if connectivity and filesystem are both ok. Such small exporter can be then reused for other services to do the same job for us. The custom exporter allows us to be more flexible in how we handle the exceptions to the availability guarantee, as we can handle them in a general purpose programming language instead of having to model them in PromQL.
+
+[cols="1,1,1,1"]
+|===
+|Requirements/Options |Exporter provided by VSHN |Exporter provided by Stackgres |Custom exporter
+
+|Checks network connectivity |✅  |✅ |✅
+
+|Login to database the same way as customer do |✅ |❌ |✅
+
+|Easily integrates with prometheus |✅ |✅ |✅
+
+|Read queries check |✅ |✅ |✅
+
+|Write queries check |❌ |❌ |✅
+
+|===
+
+=== Rationale
+
+The best solution would be to use customly written exporter, because it'll do exactly what we want with minimal footprint on resources. It disadvantage is:
```

Comment:
```suggestion
The best solution would be to write a custom exporter, because it'll do exactly what we want with minimal footprint on resources.

It disadvantage is:
```